docs/core/development/docker.md

```diff
@@ -0,0 +1,22 @@
+# Docker
```

Comment:
@jlowin What's the reasoning for moving these out of installation? I don't get the sense that the docker builds are specifically for development.

Reply:
To be honest I may have overpinned to a conversation I had in the community Slack this morning with someone who was trying to "install" Prefect via Docker and was running into some difficulty; I'm going to move it back. Thanks for pointing that out.